src/synapse.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Helper module used to manage interactions with Synapse."""
+
+import logging
+import typing
+
+import ops
+from ops.pebble import Check, ExecError
+
+from charm_state import CharmState
+from charm_types import ExecResult
+from constants import CHECK_READY_NAME, COMMAND_MIGRATE_CONFIG, COMMAND_PATH, SYNAPSE_PORT
+from exceptions import CommandMigrateConfigError
+
+logger = logging.getLogger(__name__)
+
+
+class Synapse:  # pylint: disable=too-few-public-methods
+    """A class representing the Synapse application."""
+
+    def __init__(self, charm_state: CharmState):
+        """Initialize a new instance of the Synapse class.
+
+        Args:
+            charm_state: The state of the charm that the Synapse instance belongs to.
+        """
+        self._charm_state = charm_state
+
+    def check_ready(self) -> typing.Dict:
+        """Return the Synapse container check.
+
+        Returns:
+            Dict: check object converted to its dict representation.
+        """
+        check = Check(CHECK_READY_NAME)
+        check.override = "replace"
+        check.level = "ready"
+        check.tcp = {"port": SYNAPSE_PORT}
+        # _CheckDict cannot be imported
+        return check.to_dict()  # type: ignore
+
+    def synapse_environment(self) -> typing.Dict[str, str]:
+        """Generate a environment dictionary from the charm configurations.
+
+        Returns:
+            A dictionary representing the Synapse environment variables.
+        """
+        return {
+            "SYNAPSE_SERVER_NAME": f"{self._charm_state.server_name}",
+            "SYNAPSE_REPORT_STATS": f"{self._charm_state.report_stats}",
+            # TLS disabled so the listener is HTTP. HTTPS will be handled by Traefik.
+            "SYNAPSE_NO_TLS": str(True),
```

Comment:
So we don't support E2E TLS ?
Probably not a big deal for now, but since it's a chat app supposed to be pretty secured, E2E TLS will be probably interesting to support.
I don't even know if our ingresses supports an HTTPS backend

Reply:
Not for now. I added a comment so we can check this in the future.